dev/com.ibm.ws.install/src/com/ibm/ws/install/maven/MavenDirectoryInstall.java

```diff
@@ -0,0 +1,276 @@
+		for (File esaFile : artifacts) {
+
+			logger.log(Level.FINE, Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("STATE_INSTALLING",
+					extractFeature(esaFile.getName())));
+			map.put("license.accept", true);
+			map.put("action.install", esaFile);
+			if (toExtension != null) {
+				map.put("to.extension", toExtension);
+				logger.log(Level.FINE, "Installing to extension: " + toExtension);
+			}
+
+			Integer ac = (Integer) map.get("action.result");
+			logger.log(Level.FINE, "action.result:" + ac);
+			logger.log(Level.FINE, "action.error.message:" + map.get("action.error.message"));
+
+
+			if (map.get("action.error.message") != null) {
+				// error with installation
+				logger.log(Level.FINE, "action.exception.stacktrace: " + map.get("action.error.stacktrace"));
+				String exceptionMessage = (String) map.get("action.error.message");
+				throw new InstallException(exceptionMessage);
+			} else if ((currentReturnResult = (Collection<String>) map.get("action.install.result")) != null) {
+				// installation was successful
+				if (!currentReturnResult.isEmpty()) {
+					actionReturnResult.addAll((Collection<String>) map.get("action.install.result"));
+					logger.log(Level.INFO,
+							(Messages.INSTALL_KERNEL_MESSAGES
+									.getLogMessage("LOG_INSTALLED_FEATURE", String.join(", ", currentReturnResult))
+									.replace("CWWKF1304I: ", ""))); // TODO: come up with new message for successfully
+																	// installed feature
+
+				}
+			}
+		}
+		logger.log(Level.INFO, Messages.INSTALL_KERNEL_MESSAGES.getLogMessage("TOOL_INSTALLATION_COMPLETED"));
+	}
+
+
+	private Collection<File> findEsas(Collection<String> resolvedFeatures, File rootDir) throws InstallException {
+		Collection<File> foundEsas = new HashSet<>();
+		Collection<String> featuresClone = new ArrayList<>(resolvedFeatures);
+		
+		for (String feature : resolvedFeatures) {
+			logger.log(Level.FINE, "Processing feature: " + feature);
+			File groupDir;
+			String mavenCoordinate = feature.split(":")[0];
+			String featureName = feature.split(":")[1];
+
+			if (mavenCoordinate.equals(OPEN_LIBERTY_GROUP_ID)) {
+				groupDir = new File(rootDir, "io/openliberty/features/");
+			} else if (mavenCoordinate.equals(CLOSED_LIBERTY_GROUP_ID)) {
+				groupDir = new File(rootDir, "com/ibm/websphere/appserver/features/");
+			} else {
+				throw new InstallException("Invalid maven coordinate");
+			}
+
+			Path featurePath = Paths.get(groupDir.getAbsolutePath().toString(), featureName, openLibertyVersion,
+					featureName + "-" + openLibertyVersion + ".esa");
+			logger.log(Level.FINE, featurePath.toString());
+			if (Files.isRegularFile(featurePath)) {
+				foundEsas.add(featurePath.toFile());
+				featuresClone.remove(feature);
+			}
+			
+		}
+		logger.log(Level.FINE, "ESAs not found:" + featuresClone);
+		if (!featuresClone.isEmpty()) {
+			throw new InstallException("Could not find ESA files for: " + featuresClone);
```

Comment:
Need translated message

Reply:
This is the entry point to @hermankailey artifact downloader. The translated message will be handled by the artifact downloader

Reply:
As a caller, do you need to handle any error situations?

Reply:
I'm not sure if we need to handle any error situations if the artifact downloader fails to work. If  the artifact downloader cant retreive all the missing ESA's, then we should just stop install utility right there. There is no need to uninstall the previous features because we dont start installation until we find all the ESA files to begin with

Reply:
How about the tmp?  Do we always clean it up?

Reply:
Yes